helper-cli/src/main/kotlin/common/Utils.kt

```diff
@@ -740,6 +742,20 @@ internal fun Collection<RuleViolationResolution>.mergeRuleViolationResolutions(
     return result.values.toList()
 }
 
+/**
+ * Merge the given [VulnerabilityResolution]s replacing entries with equal [VulnerabilityResolution.id].
+ */
+internal fun Collection<VulnerabilityResolution>.mergeVulnerabilityResolutions(
+    other: Collection<VulnerabilityResolution>
+): List<VulnerabilityResolution> {
+    val result = mutableMapOf<String, VulnerabilityResolution>()
+
+    associateByTo(result) { it.id }
+    other.associateByTo(result) { it.id }
```

Comment:
Hmm, can there be the case that same IDs refer to different vulnerabilities? If not, I do not see much of a problem, since you want to waive a vulnerability due to what it relevant for and not because of the URL or score.
Although this raises the question whether to offer a mechanism to waive if a vulnerability falls below a certain score.

Reply:
You are right. Currently, the only information relevant is the ID.